pandas/core/internals/concat.py

```diff
@@ -443,7 +443,7 @@ def _is_uniform_join_units(join_units: List[JoinUnit]) -> bool:
     #  cannot necessarily join
     return (
         # all blocks need to have the same type
-        all(type(ju.block) is type(join_units[0].block) for ju in join_units)
+        all(isinstance(ju.block, type(join_units[0].block)) for ju in join_units)
```

Comment:
This change is actually not equivalent. We have both ExtensionBlock as other Block types that are subclasses from ExtensionBlock.

Reply:
@jorisvandenbossche Thanks, in the first iteration of the PR, I actually retained the exact type check and opted for a `#  noqa`.  Should this be a new issue?  We could also require this behavior in a test.

Reply:
I was actually just looking at this. In the meantime it also seems that the location where this `_is_uniform_join_units` is used isn't that strict anymore about the exact block types (due to https://github.com/pandas-dev/pandas/pull/33486), so it should be fine as is.